sdv/cag/base.py

```diff
@@ -91,6 +95,61 @@ def get_updated_metadata(self, metadata):
     def _fit(self, data, metadata):
         raise NotImplementedError
 
+    def _fit_constraint_column_formatters(self, data):
+        """Fit formatters for columns that are dropped by constraints before data processing."""
+        updated_metadata = self._get_updated_metadata(self.metadata)
+        for table_name in self.metadata.tables:
+            self._constraint_col_formatters[table_name] = {}
```

Comment:
It doesn't seem like we ever need to get all the formatters for a table, so we could flatten this dictionary and just make the key a tuple of table and column name

Reply:
I think it's fine keeping it a nested dict structure, especially because we need to iterate over all the tables to reorder the columns (to ensure the data has the same column order as the one of the metadata):
https://github.com/sdv-dev/SDV/blob/bc73ea3dc4d99798c31287b8e98fbc3c29ecd58b/sdv/cag/base.py#L141